haskell/experimental/private/module.bzl

```diff
@@ -147,6 +147,11 @@ def _build_haskell_module(
     # Construct compiler arguments
 
     args = ctx.actions.args()
+
+    if (moduleAttr.src == getattr(ctx.attr, "main_file", None)
+            and getattr(ctx.attr, "main_function", None)):
+        args.add_all(["-main-is", ctx.attr.main_function])
```

Comment:
Reading the documentation of [main_function](https://api.haskell.build/haskell/defs.html#haskell_binary-main_function) looks like `-main-is` should be also passed when `main_file` is unset and the module name matches the name qualifier in `main_function` or `Main` if there is no qualifier.

Otherwise, the documentation would need to be updated.

Reply:
Should I just do `args.add_all(["-main-is", ctx.attr.main_function])` without checking that the `moduleAttr.src` matches the main_file (so its added to all the modules while compiling for a binary), because I don't know how to construct an input file target similar to `moduleAttr.src` from the module name.

Reply:
Maybe
```python
main_file = getattr(ctx.attr, "main_file", None)
main_function = getattr(ctx.attr, "main_function", None)
module_path = paths.split_extension(get_module_path_from_target(module))[0]
if main_function and (moduleAttr.src == main_file or main_function.replace(".", "/") == module_name):
    args.add_all(["-main-is", ctx.attr.main_function])
```
`get_module_path_from_target` is supposed to return "A/B/C.hs" for a module named "A.B.C".

I think this implements what the documentation says, but I'd feel better if someone double-checked it.

Reply:
Looks like this should work, however in my test setup `module_name` is an empty string for some modules but not others need to investigate what's happening.